src/location/util.rs

```diff
@@ -421,42 +929,141 @@ impl LocationUtil {
             //                    .map(|mut row| {})
             //                    .collect()
             //            })?;
+        } else if ob_vsn_major() >= 4
+            && (sub_part_func_type.is_hash_part() || sub_part_func_type.is_key_part())
+        {
+            // parseSubPartKeyHash
+            match &mut table_entry.partition_info {
+                None => {
+                    // dont have a partition info
+                    error!("LocationUtil::fetch_first_part partition_info is None");
+                    return Err(CommonErr(
+                        CommonErrCode::PartitionError,
+                        "LocationUtil:fetch_first_part partition_info is None".to_owned(),
+                    ));
+                }
+                Some(_) => match LocationUtil::parse_key_hash_part(conn, sql, table_entry, true) {
+                    Ok(v) => {
+                        let info = table_entry.partition_info.as_mut().unwrap();
+                        info.set_tablet_id_map(v)
+                    }
+                    Err(e) => {
+                        error!(
+                            "LocationUtil::fetch_first_part parse_key_hash_part error:{}",
+                            e
+                        );
+                        return Err(CommonErr(
+                            CommonErrCode::PartitionError,
+                            format!(
+                                "LocationUtil::fetch_first_part parse_key_hash_part error:{}",
+                                e
+                            ),
+                        ));
+                    }
+                },
+            }
         }
         Ok(())
     }
 
+    /*
+     * parse_key_hash_part parse tablet_id_map from sql result and return
+     */
```

Comment:
Follow the style of the rust doc comment.

Reply:
Fixed